src/renderer/app.jsx

```diff
@@ -92,16 +92,16 @@ export default class App extends React.Component {
   }
 
   saveSettings(settings) {
-    this.setState({
-      investSettings: settings,
-    });
-
     saveSettingsStore(settings);
+    ipcRenderer.invoke(
+      ipcMainChannels.SET_LANGUAGE, settings.language
+    );
+    this.setState({ investSettings: settings });
```

Comment:
I'm not sure about this. It works like this, but I would think the `setState` should only happen once the language has been set, to make sure it rerenders in the new language:
```
    ipcRenderer.invoke(
      ipcMainChannels.SET_LANGUAGE, settings.language
    ).then(() => {
        this.setState({ investSettings: settings });
    });
```
However, the tests fail like that.

Reply:
> If I understand though, it's the state of i18n that controls the language, not anything in this.state. It's just that we want to make sure a render happens after i18n.setLocale, and this.setState happens to be a way to trigger a render. But only for components whose state/props were updated. Since it's happening in app.jsx, that happens to be all components.

That all sounds right

> I could imagine a state object that did not bundle these 3 settings together in investSettings. If each were their own item in this.state, that would avoid the problem of overwriting each other.

Setting the items separately sounds like a better approach, even regardless of the overwriting issue.

> If the userEvent.selectOptions happened at the pace of a real user, we probably wouldn't have this problem.

Yeah, it works just fine when you actually click through the interface

> Do you think it's a good idea to check which setting is being updated before sending the SET_LANGUAGE call? It all seems very fast, but do you think it's worth avoiding when only the other settings were changed?

I'm not sure, but if we do refactor to set each setting individually, we could check it then?